### ISSUE TITLE
Crashes

### ISSUE THREAD

**Issue by ethanparkerr** [remyroy/CDDA-Game-Launcher#105]
with version 1.3.9 I am getting several different errors when downloading mods

```
Type: <class 'UnboundLocalError'>
Value: local variable 'archive_exception' referenced before assignment
Traceback:
File "cddagl\ui.py", line 6957, in download_http_finished
```

```
Type: <class 'FileNotFoundError'>
Value: [Errno 2] No such file or directory: '*:\\TEMP\\CDDA Game Launcher\\newmod-56bd9210\\ZSFixed_CDDAXP_2016-10-15.zip'
Traceback:
File "cddagl\ui.py", line 6948, in download_http_finished

File "zipfile.py", line 923, in __init__
```

```
Type: <class 'FileNotFoundError'>
Value: [Errno 2] No such file or directory: "*:\\TEMP\\CDDA Game Launcher\\newmod-46dc36d0\\Artyoms' Gun Emporium.rar"
Traceback:
File "cddagl\ui.py", line 6948, in download_http_finished

File "site-packages\rarfile.py", line 491, in __init__
File "site-packages\rarfile.py", line 760, in _parse
File "site-packages\rarfile.py", line 767, in _parse_real
File "site-packages\rarfile.py", line 1753, in __init__
```

**Comment by remyroy** [comment]
Are those files present on your disk when you get the error?

**Comment by ethanparkerr** [comment]
no

**Comment by remyroy** [comment]
What is the value of your TEMP environment variable? You can find out by opening a command prompt and typing `echo %TEMP%`.

**Comment by ethanparkerr** [comment]
I modified that part when pasting, originally D:\TEMP\

**Comment by remyroy** [comment]
I'm adding more validation and error messages in the whole process but that is unlikely to solve those issue. It will just prevent the launcher from crashing.

**Comment by ethanparkerr** [comment]
what's the cause? it didn't crash with the previous version so I thought it would be simple to fix

**Comment by remyroy** [comment]
I don't know. It is not crashing for me so it's hard to find out.

**Comment by remyroy** [comment]
Those crashes should be fixed with 0b90e43988f931cefa5971e7ef19a0cedef3a4d3. It will be part of the next release.

**Comment by ethanparkerr** [comment]
Crashes are fixed. and I have found the problem, dropbox domain was set to 0.0.0.0 on my hosts file so it was a dns problem. when I removed that it worked
